Sources/Widgets/Widgets3D/ResliceCursorWidget/index.js

```diff
@@ -0,0 +1,562 @@
+      focalPoint,
+      estimatedCameraPosition,
+      model.widgetState.getCenter(),
+      normal
+    );
+    const newFocalPoint = intersection.x;
+
+    renderer
+      .getActiveCamera()
+      .setFocalPoint(newFocalPoint[0], newFocalPoint[1], newFocalPoint[2]);
+
+    const newCameraPosition = [
+      newFocalPoint[0] + distance * normal[0],
+      newFocalPoint[1] + distance * normal[1],
+      newFocalPoint[2] + distance * normal[2],
+    ];
+
+    renderer
+      .getActiveCamera()
+      .setPosition(
+        newCameraPosition[0],
+        newCameraPosition[1],
+        newCameraPosition[2]
+      );
+
+    // // Renderer may not have yet actor bounds
+    const rendererBounds = renderer.computeVisiblePropBounds();
+    const bounds = model.image.getBounds();
+    rendererBounds[0] = Math.min(bounds[0], rendererBounds[0]);
+    rendererBounds[1] = Math.max(bounds[1], rendererBounds[1]);
+    rendererBounds[2] = Math.min(bounds[2], rendererBounds[2]);
+    rendererBounds[3] = Math.max(bounds[3], rendererBounds[3]);
+    rendererBounds[4] = Math.min(bounds[4], rendererBounds[4]);
+    rendererBounds[5] = Math.max(bounds[5], rendererBounds[5]);
+
+    // Don't clip away any part of the data.
+    renderer.resetCameraClippingRange(rendererBounds);
+  }
+
+  // --------------------------------------------------------------------------
+  // initialization
+  // --------------------------------------------------------------------------
+
+  model.behavior = widgetBehavior;
+  model.widgetState = stateGenerator();
+
+  publicAPI.getRepresentationsForViewType = (viewType) => {
+    switch (viewType) {
+      case ViewTypes.AXIAL:
+        return [
+          {
+            builder: vtkResliceCursorContextRepresentation,
+            labels: ['axisXinZ', 'axisYinZ'],
+            initialValues: {
+              axis1Name: 'AxisXinZ',
+              axis2Name: 'AxisYinZ',
+              viewName: 'Axial',
+            },
+          },
+        ];
+      case ViewTypes.CORONAL:
+        return [
+          {
+            builder: vtkResliceCursorContextRepresentation,
+            labels: ['axisXinY', 'axisZinY'],
+            initialValues: {
+              axis1Name: 'AxisXinY',
+              axis2Name: 'AxisZinY',
+              viewName: 'Coronal',
+            },
+          },
+        ];
+      case ViewTypes.SAGITTAL:
+        return [
+          {
+            builder: vtkResliceCursorContextRepresentation,
+            labels: ['axisYinX', 'axisZinX'],
+            initialValues: {
+              axis1Name: 'AxisYinX',
+              axis2Name: 'AxisZinX',
+              viewName: 'Sagittal',
+            },
+          },
+        ];
+      case ViewTypes.DEFAULT:
+      case ViewTypes.GEOMETRY:
+      case ViewTypes.SLICE:
+      case ViewTypes.VOLUME:
+      default:
+        return [];
+    }
+  };
+
+  publicAPI.setImage = (image) => {
+    model.widgetState.setImage(image);
```

Comment:
should this really go in widgetState ?

Reply:
I need the image's bounds when I update the state. Maybe only storing the bounds in the state is enough.

Reply:
Well, I also need the center, so I'll keep the image only in the state.

Reply:
Yes, I think you must keep only bounds. 
Center can be computed with vtkBoundingBox (or vtkBox)

Reply:
Is it relevant to recompute the center each time we update the state ?

Reply:
I didn't know that it was recomputed. So let's store only the bounds.

Reply:
Me neither :-)

However, we tend to not store the same (or derived) information multiple times. It typically leads to synchronization problems/mistakes.

Reply:
New information after rechecking. I also need the spacing of the image. What's the best ? Storing the whole image or storing only bounds and spacing ?

Reply:
I'll keep the image for now.